src/main/java/com/google/api/codegen/config/FlatteningConfig.java

```diff
@@ -41,12 +40,37 @@ public static FlatteningConfig createFlattening(
       DiagCollector diagCollector, FlatteningConfigProto flattening, Method method) {
     boolean missing = false;
     ImmutableList.Builder<ImmutableList<Field>> flatteningGroupsBuilder = ImmutableList.builder();
+    Set<Oneof> seenOneofs = new HashSet<>();
     for (FlatteningGroupProto flatteningGroup : flattening.getGroupsList()) {
       ImmutableList.Builder<Field> parametersBuilder = ImmutableList.builder();
       for (String parameter : flatteningGroup.getParametersList()) {
         Field parameterField = method.getInputMessage().lookupField(parameter);
         if (parameterField != null) {
           parametersBuilder.add(parameterField);
+          Oneof oneof = parameterField.getOneof();
+          if (oneof != null) {
+            diagCollector.addDiag(
+                Diag.warning(
+                    SimpleLocation.TOPLEVEL,
+                    "Oneof field flattened: method = %s, message type = %s, oneof = %s, field = %s",
+                    method.getFullName(),
+                    method.getInputMessage().getFullName(),
+                    oneof.getName(),
+                    parameter));
```

Comment:
There is difference between flattening and required/optional: it is a possible approach in languages that support overriding (and therefore flattening) to provide n methods if there are n possible values of the oneof; that is, if you have a request

```
FooRequest {
  oneof {
    A a = 1;
    B b = 2;
    C c = 3;
  }
}
```

You could potentially multiple overloaded methods

```
function foo(A a) { ... }
function foo(B b) { ... }
function foo(C c) { ... }
```

In fact, this is currently how `RunQuery` for datastore is currently configured: https://github.com/googleapis/googleapis/blob/master/google/datastore/v1/datastore_gapic.yaml#L52

I think this is a bad idea because it only works if no two oneof values have the same type. But it's not universally illegal as it would be to do this in a language that doesn't support overloading (and therefore uses the required parameters configuration).